gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py

```diff
@@ -96,13 +96,69 @@ def get_ovr_count(filename_or_ds: PathOrDS):
         return bnd.GetOverviewCount()
 
 
-def get_ovr_idx(filename_or_ds: PathOrDS, ovr_idx: Optional[int]):
+def get_sizes_factors_resolutions(filename_or_ds: PathOrDS, dim: Optional[int]=0):
+    ds = open_ds(filename_or_ds)
+    bnd = ds.GetRasterBand(1)
+    ovr_count = bnd.GetOverviewCount()
+    geo_transform = ds.GetGeoTransform()
+    r0 = geo_transform[1], geo_transform[5]
+    s0 = ds.RasterXSize, ds.RasterYSize
+    f0 = 1, 1
+    sizes = [s0]
+    factors = [f0]
+    resolutions = [r0]
+    for i_overview in range(ovr_count):
+        h_overview = bnd.GetOverview(i_overview)
+        if h_overview is not None:
+            s = h_overview.XSize, h_overview.YSize
+            f = s0[0] / s[0], s0[1] / s[1]
+            r = r0[0] * f[0], r0[1] * f[1]
+            sizes.append(s)
+            factors.append(f)
+            resolutions.append(r)
+    if dim is not None:
+        sizes = [x[dim] for x in sizes]
+        factors = [x[dim] for x in factors]
+        resolutions = [x[dim] for x in resolutions]
+    return sizes, factors, resolutions
+
+
+def get_best_ovr_by_resolutions(requested_res: float, resolutions: Sequence[float]):
+    for ovr, res in enumerate(resolutions):
+        if res > requested_res:
+            return max(0, ovr-1)
+    return len(resolutions)-1
+
+
+def get_ovr_idx(filename_or_ds: PathOrDS,
+                ovr_idx: Optional[Union[int, float]] = None,
+                ovr_res: Optional[Union[int, float]] = None):
+    """
+    returns a non-negative ovr_idx, from given mutually exclusive ovr_idx (index) or ovr_res (resolution)
+    ovr_idx == None and ovr_res == None => returns 0
+    ovr_idx: int >= 0 => returns the given ovr_idx
+    ovr_idx: int < 0 => -1 is the last overview; -2 is the one before the last and so on
+    ovr_res: float|int => returns the best suitable overview for a given resolution
+             meaning the ovr with the lowest resolution which is higher then the request
+    ovr_idx: float = x => same as (ovr_idx=None, ovr_res=x)
```

Comment:
This possibility is really messy. Why keep it ?
The below code should really distinguish the case where ovr_idx or over_res are specified and not mix things

Reply:
hmm.. because it simplifies my code - I have some calls to this function deep inside some inner functions and I pass ovr_idx from the outer functions, now I wanted to add support for choosing overview by resolution. The easiest solution is to pass a float to the outer function, that's all the change I needed to make. Otherwise if I wanted to use two parameters I would need to modify all the functions on the way. Also having mutually exclusive parameters is more error prone in my opinion.
I agree that it might be confusing so I added both options (single or double parameters). 
If you think it is less messy - I don't mind changing it to be an undocumented feature and to change the type hinting to suggest ovr_idx: Optional[int] accordingly.
If you think about it, the implementation also becomes much simpler if you use this approach of a single parameter - less cases to test against.

Reply:
> If you think it is less messy - I don't mind changing it to be an undocumented feature

Yes, it would best not to advertize ovr_idx: float in the documentation of the function. In the implementation, you could still mention that subtelty.

> If you think about it, the implementation also becomes much simpler if you use this approach of a single parameter - less cases to test against.

Hum the testing should be little impacted. Testing the case where a parameter is integer or float, or two different parameters is pretty much equivalent

Reply:
done